### ISSUE TITLE
CLI 2 critical severity vulnerabilities

### ISSUE THREAD

**Issue by bourpie** [usebruno/bruno#922]
When installing CLI on Node V. 18.14.1, I get this message

```
npm WARN deprecated vm2@3.9.19: The library contains critical security issues and should not be used for production! The maintenance of the project has been discontinued. Consider 
migrating your code to isolated-vm.

changed 146 packages, and audited 147 packages in 11s

34 packages are looking for funding
  run `npm fund` for details

2 critical severity vulnerabilities
```

**Comment by Its-treason** [comment]
`vm2` has been deprecated for a while now, and we already planned to update to a different library here #263.

The vulnerabilities in `vm2` allow for malicious code to escape the sandbox. I think if you only run trusted request scripts it should be fine for now, but we will definite address this in the future.

**Comment by chriswarkentin** [comment]
I would say that this should be fairly high priority if you're looking to drive adoption. I'm evaluating Bruno to use at my workplace and things like that will definitely raise eyebrows.

**Comment by camba1** [comment]
Is there any progress on this? I love Bruno, but unfortunately cannot use the CLI until this is addressed and thus cannot use it to automate API testing

**Comment by Its-treason** [comment]
> Is there any progress on this? I love Bruno, but unfortunately cannot use the CLI until this is addressed and thus cannot use it to automate API testing

Not really, I opened a second PR that replaced vm2 with the native vm module. But this did not get any attention yet.

**Comment by qury** [comment]
Are there any plans from the main contributors to fix any of the critical bugs?

As of today this is what i get from npm audit:
`44 vulnerabilities (31 moderate, 9 high, 4 critical)`

Critical packages being:
- vm2
  - vm2 Sandbox Escape vulnerability - https://github.com/advisories/GHSA-cchq-frgv-rjh5
  - vm2 Sandbox Escape vulnerability - https://github.com/advisories/GHSA-g644-9gfx-q4q4

- underscore
   - Arbitrary Code Execution in underscore - https://github.com/advisories/GHSA-cf4h-3jhx-xvhq
- minimist
   - Prototype Pollution in minimist - https://github.com/advisories/GHSA-vh95-rmgr-6w4m
   - Prototype Pollution in minimist - https://github.com/advisories/GHSA-xvch-5gv4-984h

**Comment by treynolds011** [comment]
I know that there has been some discussion of getting this fixed but I would love to start using the cli in my environment but can't until this is fixed.

**Comment by cemerick1** [comment]
I agree with the above comments about this getting fixed so we can use it in our CI/CD pipelines.

**Comment by tonytvo** [comment]
it's very critical that we can get command line for bruno working for CI/CD as CI/CI is very critical components in software development and will definitely drive adoption for bruno. It would be greatly appreciated, it we could get this fix soon, it stops me from using it in my workplace

**Comment by brvaland** [comment]
Is it possible to share timelines when this issue will be resolved ?

This is critical security vulnerability so we can use bru cli in cd/ci pipelines.

**Comment by helloanoop** [comment]
We recently upgraded our libraries and the critical vulnerabilities mentioned in this ticket has been fixed.
1 new critical vulnerability is shown in the audit log. This is because we are using `crypto-js@^3.1.9-1` which is needed to access crypto functions inside scripting [Safe Mode](https://docs.usebruno.com/get-started/javascript-sandbox)

This is not an easy fix since, we use [quickjs](https://bellard.org/quickjs/) as the JS execution engine inside [Safe Mode](https://docs.usebruno.com/get-started/javascript-sandbox). QuickJS does not provide native encryption utils like node vm (which is used in developer mode)

We are working on implementing [shims](https://github.com/usebruno/bruno/tree/main/packages/bruno-js/src/sandbox/quickjs/shims/lib) to make this work in Safe mode using the latest crypto-js library

<img width="686" alt="image" src="https://github.com/user-attachments/assets/e600a977-8795-42dc-bc34-5a753ac2e40d">

**Comment by helloanoop** [comment]
We were unable to create [shims](https://github.com/usebruno/bruno/tree/main/packages/bruno-js/src/sandbox/quickjs/shims/lib) for crypto-js to make it compatible with [Safe Mode](https://docs.usebruno.com/get-started/javascript-sandbox).

To resolve this, we forked [crypto-js](https://github.com/usebruno/crypto-js) specifically for Safe Mode usage, while Developer Mode continues to utilize the latest version of crypto-js. From a security perspective, I feel this approach is acceptable for now, as the crypto module is only used for testing purposes (inside Bruno) and not on any public-facing servers.

With all outstanding vulnerabilities addressed, I am closing this issue.
